script/tool/lib/src/version_check_command.dart

```diff
@@ -205,16 +200,63 @@ class VersionCheckCommand extends PluginCommand {
                 NextVersionType.BREAKING_MAJOR) {
           final String error = '$pubspecPath breaking change detected.\n'
               'Breaking changes to platform interfaces are strongly discouraged.\n';
-          final Colorize redError = Colorize(error)..red();
-          print(redError);
-          throw ToolExit(1);
+          ThrowsToolExit(errorMessage: error);
         }
       } on io.ProcessException {
         print('Unable to find pubspec in master for $pubspecPath.'
             ' Safe to ignore if the project is new.');
       }
     }
 
+    await for (Directory plugin in getPlugins()) {
+      await _checkVersionsMatch(plugin);
+    }
+
     print('No version check errors found!');
   }
+
+  Future<void> _checkVersionsMatch(Directory plugin) async {
+    // get version from pubspec
+    final String packageName = plugin.basename;
+    print('================================================================');
+    print('Checking that $packageName has matching version in CHANGELOG and pubspec');
+
+    final Pubspec pubspec = _tryParsePubspec(plugin);
+    final Version fromPubspec = pubspec.version;
+
+    // get version from CHANGELOG
+    final File changelog = plugin.childFile('CHANGELOG.md');
+    final List<String> lines = changelog.readAsLinesSync();
+    final String firstLine = lines.first;
+    // Remove all leading mark down syntax from the version line.
+    final String versionString = firstLine.split(' ').last;
+    Version fromChangeLog = Version.parse(versionString);
+    if (fromChangeLog == null) {
+      final String error = 'Can not find version on the first line of CHANGELOG.md';
+      ThrowsToolExit(errorMessage: error);
+    }
+
+    if (fromPubspec != fromChangeLog) {
+      final String error = 'versions for $packageName in CHANGELOG.md and pubspec.yaml do not match.';
+      ThrowsToolExit(errorMessage: error);
+    }
+    print('${plugin.basename} passed version check');
+  }
+
+  // ignore: missing_return
```

Comment:
Isn't this a real bug you're suppressing, since you're only handling `Exception`?